### PR TITLE
fix: reinstate gc test

### PR DESCRIPTION
**Pull request by mrekucci**
This change is [Reviewable](https://reviewable.io/reviews/ethersphere/bee/2318)

### CODE REVIEW

**Review by zelig — APPROVED** [review]
> Reviewed 7 of 7 files at r1.
> *[Reviewable](https://reviewable.io/reviews/ethersphere/bee/2318)* status: :shipit: complete! all files reviewed, all discussions resolved (waiting on @acud and @vandot)

File: .github/workflows/beekeeper.yml [hunk 1]
```diff
@@ -44,6 +44,8 @@ jobs:
           patch pkg/postage/postagecontract/contract.go .github/patches/postagecontract.patch
           patch pkg/postage/listener/listener.go .github/patches/listener.patch
           patch pkg/postage/service.go .github/patches/postageservice.patch
+          patch pkg/postage/batchstore/reserve.go .github/patches/postagereserve_gc.patch
```
Comment (acud) [inline .github/workflows/beekeeper.yml:47]: this needs to happen before the storage incentives setup is built/started. have a look how it was before we took the test out of the pipeline

**Review by mrekucci — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/ethersphere/bee/2318)* status: 4 of 7 files reviewed, 1 unresolved discussion (waiting on @acud, @vandot, and @zelig)

___
*[.github/workflows/beekeeper.yml, line 47 at r4](https://reviewable.io/reviews/ethersphere/bee/2318#-Me59CIfIEqzd9MqALhi-r4-47:-Mf71CkzEiU0qjewzYzL:b-896fix) ([raw file](https://github.com/ethersphere/bee/blob/3085bf5419e015618e501eed7c0990167ab3826b/.github/workflows/beekeeper.yml#L47)):*
<details><summary><i>Previously, acud (acud) wrote…</i></summary><blockquote>

this needs to happen before the storage incentives setup is built/started. have a look how it was before we took the test out of the pipeline
</blockquote></details>

Done.

**Review by acud — APPROVED** [review]
:lgtm:

> Reviewed 4 of 7 files at r1, 1 of 1 files at r2, 1 of 2 files at r4, 1 of 1 files at r6.
> *[Reviewable](https://reviewable.io/reviews/ethersphere/bee/2318#-:-Mf7C-_z22PHxY8NzvWq:bnfp4nl)* status: all files reviewed, 1 unresolved discussion (waiting on @mrekucci and @vandot)

___
*[.github/workflows/beekeeper.yml, line 134 at r6](https://reviewable.io/reviews/ethersphere/bee/2318#-Mf7Bnth8J9muNmGBLsU:-Mf7Bnth8J9muNmGBLsV:beg4nft) ([raw file](https://github.com/ethersphere/bee/blob/949f6266cb98a2522676e6fed7c5ad160831f6a7/.github/workflows/beekeeper.yml#L134)):*
> ```YAML
>       - name: Test gc
>         id: gc-chunk-1
>         run: beekeeper check --cluster-name local-gc --checks=ci-gc
> ```

nit, consider adding a `destroy cluster` step after this one

**Pull request merged**